### PR TITLE
Do not use config subscription in couch_index

### PR DESCRIPTION
**Pull request by iilyak**
The simple fact that we call config:subscribe_for_changes causes the hike in memory consumed by couch_index process. Monitoring the changes for "query_server_config.commit_freq" is not worth it. We would just call config:get_index when we need to. 

COUCHDB-3135

### CODE REVIEW

File: src/couch_index.erl [hunk 1]
```diff
@@ -30,16 +29,14 @@
...
--define(CONFIG_SUBSCRIPTION, [{"query_server_config", "commit_freq"}]).
 
 -record(st, {
     mod,
     idx_state,
     updater,
     compactor,
     waiters=[],
-    commit_delay,
+    commit_delay, %% not used anymore
```
Comment (davisp) [inline src/couch_index.erl]: You should remove it then.

**Comment by davisp** [comment]
+1 after removing the unused record field.

**Comment by rnewson** [comment]
+1

**Comment by rnewson** [comment]
(can be merged for 2.0)

**Pull request merged**